docs/proposals/PROPOSAL_E.md

```diff
@@ -184,6 +189,36 @@ For registries that do not support the `referrers` API, a tag MUST be pushed for
 - Periodic garbage collection may be performed by clients pushing new referrers, deleting stale referrers that have been replaced with newer versions, and tags that no longer point to an accessible manifest.
 - Clients can verify the registry does not support the `referrers` API by querying the API and checking for a 404.
 
+#### Client Expectations
+
+##### Creating Artifacts
+
+- For portability, clients should generate artifacts using image-spec until all registries where the artifact is pushed to have been upgraded.
+- The descriptor of the manifest this artifact refers to should be set in the `refers` field of the manifest.
+- Other metadata should be included in the manifest annotations.
+- When pushing, the registry should be checked for the referrers API support.
+- If the referrers API is not available, the artifact should be pushed with a tag using the digest schema. Tags with the digest schema should not be pushed if the referrers API is available.
+
+##### Pulling Artifacts
+
+To pull artifacts that reference an existing manifest:
+
+- Clients requests artifacts associated with a manifest first query the referrers API using the digest of the requested manifest.
+- If the request fails, clients must fall back to listing tags in the repository and pull manifests for each tag with the matching `<alg>` and `<ref>` prefix.
```

Comment:
Can't clients avoid the tag listing API if they know the digest-type of the tag they need ?
Since we have this capabilty with tags I would say it would be worth considering if we can add the artifactType filter to the API as well.

Reply:
If clients know the digest of the artifact (the value of `<hash>`), they can skip the tag listing, but I don't know how they'd get that.